core/src/main/java/org/elasticsearch/index/shard/IndexShard.java

```diff
@@ -512,33 +512,32 @@ private IndexShardState changeState(IndexShardState newState, String reason) {
     }
 
     /**
-     * Index a document and return whether it was created, as opposed to just
-     * updated.
+     * Index a document
      */
-    public boolean index(Engine.Index index) {
+    public void index(Engine.Index index) {
         ensureWriteAllowed(index);
         Engine engine = getEngine();
-        return index(engine, index);
+        index(engine, index);
     }
 
-    private boolean index(Engine engine, Engine.Index index) {
+    private void index(Engine engine, Engine.Index index) {
         active.set(true);
         index = indexingOperationListeners.preIndex(index);
-        final boolean created;
+        if (logger.isTraceEnabled()) {
+            logger.trace("index [{}][{}]{}", index.type(), index.id(), index.docs());
+        }
         try {
-            if (logger.isTraceEnabled()) {
-                logger.trace("index [{}][{}]{}", index.type(), index.id(), index.docs());
-            }
-            created = engine.index(index);
+            engine.index(index);
             index.endTime(System.nanoTime());
+            if (index.hasFailure()) {
```

Comment:
can we do this out of the try catch? now an exception in postIndex will cause it to be called twice no?